src/pretty_printer.ts

```diff
@@ -3,28 +3,41 @@ import { isPhrasingTag } from "./content_categories.js";
 import { BaseHTMLElement, Doctype, type HTMLDocument, VoidBaseHTMLElement } from "./html_element.js";
 import { VoidXMLElement, XMLDeclaration, type XMLDocument, XMLElement } from "./xml.js";
 
-const escapeStr = (text: string): string => {
+const escapeXml = (text: string): string => {
 	return stringifyEntities(text, { escapeOnly: true });
 };
 
+const ESCAPE_HTML = {
+	"&": "&amp;",
+	"<": "&lt;",
+	">": "&gt;",
+	"'": "&#39;",
+	'"': "&quot;",
+} as const;
+
+const escapeHtml = (text: string): string =>
+	text.replaceAll(/[&<>'"]/g, (match) => ESCAPE_HTML[match as keyof typeof ESCAPE_HTML]);
+
+const escapeText = (mode: FormatterMode, text: string): string => (mode === "xml" ? escapeXml(text) : escapeHtml(text));
+
```

Comment:
The [``stringify-entities`](https://www.npmjs.com/package/stringify-entities) package should handle this, probably a bug from my side. The XML got changed here and I don't think it's correctly escaped now. I'll take a closer look after work.

Reply:
I added this because I didn't like that it was escaping using non-name entities (`&amp;`, `&quot;`, etc) but it looks like we'd just need to use `useNamedReferences` for that. Also, xml escaping looks a good deal more complicated and I'm not sure how to reliably escape it correctly:

https://stackoverflow.com/a/46637835/25498506

Reply:
Let's just ignore XML for now, I really just wanted it for sitemaps and RSS/Atom feeds which I've never seen fully escaped 🙈